diff --git a/imgui.cpp b/imgui.cpp
index d8c17e43..47c49b20 100644
--- a/imgui.cpp
+++ b/imgui.cpp
@@ -14766,26 +14766,26 @@ void ImGuiEx::ShowStackToolWindow(ImGuiContext* ctx, bool* p_open)
 
 #else
 
-void ImGuiEx::ShowMetricsWindow(bool*) {}
-void ImGuiEx::ShowFontAtlas(ImFontAtlas*) {}
-void ImGuiEx::DebugNodeColumns(ImGuiOldColumns*) {}
-void ImGuiEx::DebugNodeDrawList(ImGuiWindow*, const ImDrawList*, const char*) {}
+void ImGuiEx::ShowMetricsWindow(ImGuiContext*, bool*) {}
+void ImGuiEx::ShowFontAtlas(ImGuiContext*, ImFontAtlas*) {}
+void ImGuiEx::DebugNodeColumns(ImGuiContext*, ImGuiOldColumns*) {}
+void ImGuiEx::DebugNodeDrawList(ImGuiContext*, ImGuiWindow*, const ImDrawList*, const char*) {}
 void ImGuiEx::DebugNodeDrawCmdShowMeshAndBoundingBox(ImDrawList*, const ImDrawList*, const ImDrawCmd*, bool, bool) {}
-void ImGuiEx::DebugNodeFont(ImFont*) {}
-void ImGuiEx::DebugNodeStorage(ImGuiStorage*, const char*) {}
-void ImGuiEx::DebugNodeTabBar(ImGuiTabBar*, const char*) {}
-void ImGuiEx::DebugNodeWindow(ImGuiWindow*, const char*) {}
-void ImGuiEx::DebugNodeWindowSettings(ImGuiWindowSettings*) {}
-void ImGuiEx::DebugNodeWindowsList(ImVector<ImGuiWindow*>*, const char*) {}
-void ImGuiEx::DebugNodeViewport(ImGuiViewportP*) {}
-
-void ImGuiEx::DebugLog(const char*, ...) {}
-void ImGuiEx::DebugLogV(const char*, va_list) {}
-void ImGuiEx::ShowDebugLogWindow(bool*) {}
-void ImGuiEx::ShowStackToolWindow(bool*) {}
-void ImGuiEx::DebugHookIdInfo(ImGuiID, ImGuiDataType, const void*, const void*) {}
-void ImGuiEx::UpdateDebugToolItemPicker() {}
-void ImGuiEx::UpdateDebugToolStackQueries() {}
+void ImGuiEx::DebugNodeFont(ImGuiContext*, ImFont*) {}
+void ImGuiEx::DebugNodeStorage(ImGuiContext*, ImGuiStorage*, const char*) {}
+void ImGuiEx::DebugNodeTabBar(ImGuiContext*, ImGuiTabBar*, const char*) {}
+void ImGuiEx::DebugNodeWindow(ImGuiContext*, ImGuiWindow*, const char*) {}
+void ImGuiEx::DebugNodeWindowSettings(ImGuiContext*, ImGuiWindowSettings*) {}
+void ImGuiEx::DebugNodeWindowsList(ImGuiContext*, ImVector<ImGuiWindow*>*, const char*) {}
+void ImGuiEx::DebugNodeViewport(ImGuiContext*, ImGuiViewportP*) {}
+
+void ImGuiEx::DebugLog(ImGuiContext*, const char*, ...) {}
+void ImGuiEx::DebugLogV(ImGuiContext*, const char*, va_list) {}
+void ImGuiEx::ShowDebugLogWindow(ImGuiContext*, bool*) {}
+void ImGuiEx::ShowStackToolWindow(ImGuiContext*, bool*) {}
+void ImGuiEx::DebugHookIdInfo(ImGuiContext*, ImGuiID, ImGuiDataType, const void*, const void*) {}
+void ImGuiEx::UpdateDebugToolItemPicker(ImGuiContext*) {}
+void ImGuiEx::UpdateDebugToolStackQueries(ImGuiContext*) {}
 
 #endif // #ifndef IMGUI_DISABLE_DEBUG_TOOLS
 
diff --git a/imgui_tables.cpp b/imgui_tables.cpp
index 16935725..6bfd1ff6 100644
--- a/imgui_tables.cpp
+++ b/imgui_tables.cpp
@@ -3660,8 +3660,8 @@ void ImGuiEx::DebugNodeTableSettings(ImGuiContext* ctx, ImGuiTableSettings* sett
 
 #else // #ifndef IMGUI_DISABLE_DEBUG_TOOLS
 
-void ImGuiEx::DebugNodeTable(ImGuiTable*) {}
-void ImGuiEx::DebugNodeTableSettings(ImGuiTableSettings*) {}
+void ImGuiEx::DebugNodeTable(ImGuiContext*, ImGuiTable*) {}
+void ImGuiEx::DebugNodeTableSettings(ImGuiContext*, ImGuiTableSettings*) {}
 
 #endif
 
