diff --git a/imgui.h b/imgui.h
index feb5bbb2..0ff3d5c4 100644
--- a/imgui.h
+++ b/imgui.h
@@ -3015,7 +3015,7 @@ namespace ImGuiEx
 #ifndef IMGUI_DISABLE_OBSOLETE_KEYIO
     IMGUI_API ImGuiKey     GetKeyIndex(ImGuiContext* ctx, ImGuiKey key);  // map ImGuiKey_* values into legacy native key index. == io.KeyMap[key]
 #else
-    static inline ImGuiKey GetKeyIndex(ImGuiKey key)   { IM_ASSERT(key >= ImGuiKey_NamedKey_BEGIN && key < ImGuiKey_NamedKey_END && "ImGuiKey and native_index was merged together and native_index is disabled by IMGUI_DISABLE_OBSOLETE_KEYIO. Please switch to ImGuiKey."); return key; }
+    static inline ImGuiKey GetKeyIndex(ImGuiContext*, ImGuiKey key)   { IM_ASSERT(key >= ImGuiKey_NamedKey_BEGIN && key < ImGuiKey_NamedKey_END && "ImGuiKey and native_index was merged together and native_index is disabled by IMGUI_DISABLE_OBSOLETE_KEYIO. Please switch to ImGuiKey."); return key; }
 #endif
 }
 
