diff --git a/imgui_implicit.cpp b/imgui_implicit.cpp
index 9859623f..56ccea56 100644
--- a/imgui_implicit.cpp
+++ b/imgui_implicit.cpp
@@ -407,11 +407,5 @@ bool ImGui::BeginTabItem(const char* label, bool* p_open , ImGuiTabItemFlags fla
 void ImGui::EndTabItem() { ImGuiEx::EndTabItem(GImGui); }
 bool ImGui::TabItemButton(const char* label, ImGuiTabItemFlags flags ) { bool r = ImGuiEx::TabItemButton(GImGui, label, flags); return r; }
 void ImGui::SetTabItemClosed(const char* tab_or_docked_window_label) { ImGuiEx::SetTabItemClosed(GImGui, tab_or_docked_window_label); }
-void ImGui::ShowDemoWindow(bool* p_open ) { ImGuiEx::ShowDemoWindow(GImGui, p_open); }
-void ImGui::ShowAboutWindow(bool* p_open ) { ImGuiEx::ShowAboutWindow(GImGui, p_open); }
-void ImGui::ShowFontSelector(const char* label) { ImGuiEx::ShowFontSelector(GImGui, label); }
-bool ImGui::ShowStyleSelector(const char* label) { bool r = ImGuiEx::ShowStyleSelector(GImGui, label); return r; }
-void ImGui::ShowStyleEditor(ImGuiStyle* ref ) { ImGuiEx::ShowStyleEditor(GImGui, ref); }
-void ImGui::ShowUserGuide() { ImGuiEx::ShowUserGuide(GImGui); }
 
 #endif // IMGUI_DISABLE_IMPLICIT_API
