diff --git a/imgui_demo.cpp b/imgui_demo.cpp
index ae5ee93f..cb5276a2 100644
--- a/imgui_demo.cpp
+++ b/imgui_demo.cpp
@@ -245,10 +245,6 @@ void*                               GImGuiDemoMarkerCallbackUserData = NULL;
 // You may then search for keywords in the code when you are interested by a specific feature.
 void ImGuiEx::ShowDemoWindow(ImGuiContext* ctx, bool* p_open)
 {
-    // Exceptionally add an extra assert here for people confused about initial Dear ImGui setup
-    // Most functions would normally just crash if the context is missing.
-    IM_ASSERT(ImGuiEx::GetCurrentContext() != NULL && "Missing dear imgui context. Refer to examples app!");
-
     // Examples Apps (accessible from the "Examples" menu)
     static bool show_app_main_menu_bar = false;
     static bool show_app_documents = false;
